### PR TITLE
cip-190 deprecations

### PR DESCRIPTION
**Pull request by hvub**
Deprecation message related to

- https://github.com/neo-technology/neo4j/pull/27564
- https://github.com/neo4j/docs-status-codes/pull/202
- cf. [CIP-190](https://docs.google.com/document/d/1AONjI-SpSEla-bHPJ6ghXBfF9alN5mGiOHOTEyfBtRc/edit)

### CODE REVIEW

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-1084.surge.sh

Updated pages:
- [Deprecations, additions, and compatibility](https://neo4j-docs-cypher-1084.surge.sh/cypher-manual/5/deprecations-additions-removals-compatibility)

File: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc [hunk 1]
```diff
@@ -16,6 +16,78 @@ New features are added to the language continuously, and occasionally, some feat
...
+label:functionality[]
+label:deprecated[]
+[source, cypher, role="noheader"]
+----
+MATCH (where {...})
+----
+----
+MATCH (...)-[where {...}]->()
+----
+| The unescaped variable named `where` or `WHERE` (or in any other casing) in a node pattern or a relationship pattern if directly followed by a property key-value expression only is deprecated. Use backticks to escape the variable name: `MATCH (++`where`++ { ... })` or `MATCH (...)-[++`where`++ { ... }]->()`.
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]:
```suggestion
a| The unescaped variable named `where` (or any casing variant, like `WHERE`) used in a node or relationship pattern followed directly by a property key-value expression is deprecated.
To continue using variables with this name, use backticks to escape the variable name:

* Node patterns: `MATCH (++`where`++ { ... })`
* Relationship patterns: `MATCH (...)-[++`where`++ { ... }]->()`
```

File: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc [hunk 2]
```diff
@@ -16,6 +16,78 @@ New features are added to the language continuously, and occasionally, some feat
...
+| The unparenthesized label expression predicate as right-hand side operand of `\+` is deprecated. Parenthesize the label expression predicate on the right-hand side of `+`: `... + (n:A)`.
+
+a|
+label:functionality[]
+label:deprecated[]
+[source, cypher, role="noheader"]
+----
+CASE x ... WHEN is :: STRING THEN ... END
+----
+| The type predicate expression on an unescaped variable named `is` or `IS` (or in any other casing) as a when operand of a simple `CASE` is deprecated. Use backticks to escape the variable name: `CASE x ... WHEN ++`is`++ :: STRING THEN ... END`
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]:
```suggestion
| Using an unescaped variable named `is`(or any casing variant, like `IS`) as a `WHEN` operand in a simple xref:queries/case.adoc#case-simple[simple `CASE`] expression is deprecated.
To continue using variables with this name in simple `CASE` expressions, use backticks to escape the variable name: `CASE x ... WHEN ++`is`++ :: STRING THEN ... END`
```

File: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc [hunk 3]
```diff
@@ -16,6 +16,78 @@ New features are added to the language continuously, and occasionally, some feat
...
+label:functionality[]
+label:deprecated[]
+[source, cypher, role="noheader"]
+----
+CASE x ... WHEN contains + 1 THEN ... END
+----
+----
+CASE x ... WHEN contains - 1 THEN ... END
+----
+| The addition or subtraction on an unescaped variable named `contains` or `CONTAINS` (or in any other casing) as a when operand of a simple `CASE` is deprecated. Use backticks to escape the variable name: `CASE x ... WHEN ++`contains`++ + 1 THEN ... END` or `CASE x ... WHEN ++`contains`++ - 1 THEN ... END`
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]:
```suggestion
a| Using an unescaped variable named `contains` (or any casing variant, like `CONTAINS`) in addition or subtraction operations within a `WHEN` operand of a xref:queries/case.adoc#case-simple[simple `CASE`] expression is deprecated.
To continue using variables with this name, use backticks to escape the variable name:

* Additions: `CASE x ... WHEN ++`contains`++ + 1 THEN ... END` 
* Subtractions: `CASE x ... WHEN ++`contains`++ - 1 THEN ... END`
```

File: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc [hunk 4]
```diff
@@ -16,6 +16,78 @@ New features are added to the language continuously, and occasionally, some feat
...
+label:functionality[]
+label:deprecated[]
+[source, cypher, role="noheader"]
+----
+CASE x ... WHEN in[1] THEN ... END
+----
+----
+CASE x ... WHEN in["abc"] THEN ... END
+----
+| The `[]` operator on an unescaped variable named `in` or `IN` (or in any other casing) as a when operand of a simple `CASE` is deprecated. Use backticks to escape the variable name: `CASE x ... WHEN ++`in`++[1] THEN ... END` or `CASE x ... WHEN ++`contains`++["abc"] THEN ... END`
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]:
```suggestion
a| Using the `[]` operator on an unescaped variable named `in` (or any casing variant, like `IN`) within a `WHEN` operand of a xref:queries/case.adoc#case-simple[simple `CASE`] expression is deprecated.
To continue using variables with this name, use backticks to escape the variable name.

 * `CASE x ... WHEN ++`in`++[1] THEN ... END` 
 * `CASE x ... WHEN ++`contains`++["abc"] THEN ... END`
```

File: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc [hunk 5]
```diff
@@ -16,6 +16,78 @@ New features are added to the language continuously, and occasionally, some feat
...
+----
+... + n:A
+----
+----
+... + n:A&B
+----
+----
+... + n:A&B\|C
+----
+| The unparenthesized label expression predicate as right-hand side operand of `\+` is deprecated. Parenthesize the label expression predicate on the right-hand side of `+`: `... + (n:A)`.
```
Comment (JPryce-Aklundh) [inline modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]:
```suggestion
| Using an unparenthesized label expression predicate as the right-hand side operand of `+` is deprecated.
Parenthesize the label expression predicate on the right-hand side of `+`: `... + (n:A)`.
```

File: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc (hunk 4 above)
Comment (hvub) [reply modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]: `CASE x ... WHEN ++`contains`++["abc"] THEN ... END` was copy–paste error on my part, it should be `CASE x ... WHEN ++`in`++["abc"] THEN ... END`, of course.

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-1084.surge.sh

Updated pages:
- [Deprecations, additions, and compatibility](https://neo4j-docs-cypher-1084.surge.sh/cypher-manual/5/deprecations-additions-removals-compatibility)

**Review by JPryce-Aklundh — APPROVED** [review]

File: modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc [hunk 6]
```diff
@@ -16,6 +16,93 @@ New features are added to the language continuously, and occasionally, some feat
...
+Parenthesize the label expression predicate on the right-hand side of `+`: `... + (n:A)`.
+
+a|
+label:functionality[]
+label:deprecated[]
+[source, cypher, role="noheader"]
+----
+CASE x ... WHEN is :: STRING THEN ... END
+----
+a| Using an unescaped variable named `is`(or any casing variant, like `IS`) as a `WHEN` operand in a simple xref:queries/case.adoc#case-simple[simple `CASE`] expression is deprecated.
```
Comment (hvub) [inline modules/ROOT/pages/deprecations-additions-removals-compatibility.adoc]:
```suggestion
a| Using an unescaped variable named `is` (or any casing variant, like `IS`) as a `WHEN` operand in a xref:queries/case.adoc#case-simple[simple `CASE`] expression is deprecated.
```

**Comment by neo-technology-commit-status-publisher** [comment]
This PR includes documentation updates
View the updated docs at https://neo4j-docs-cypher-1084.surge.sh

Updated pages:
- [Deprecations, additions, and compatibility](https://neo4j-docs-cypher-1084.surge.sh/cypher-manual/5/deprecations-additions-removals-compatibility)

**Review by loveleif — APPROVED** [review]

**Comment by neo-technology-commit-status-publisher** [comment]
Thanks for the documentation updates.

The preview documentation has now been torn down - reopening this PR will republish it.

**Pull request merged**